js_modules/dagster-ui/packages/ui-core/src/partitions/DimensionRangeInput.tsx

```diff
@@ -18,12 +18,18 @@ export const DimensionRangeInput = ({
   isTimeseries: boolean;
 }) => {
   const [valueString, setValueString] = React.useState('');
+
+  const valueJSON = React.useMemo(() => JSON.stringify(value), [value]);
   const partitionNameJSON = React.useMemo(() => JSON.stringify(partitionKeys), [partitionKeys]);
 
   React.useEffect(() => {
+    // Only reset the valueString if the valueJSON meaningfully changes
     const partitionNameArr = JSON.parse(partitionNameJSON);
-    setValueString(isTimeseries ? partitionsToText(value, partitionNameArr) : value.join(', '));
-  }, [value, partitionNameJSON, isTimeseries]);
+    const valueArr = JSON.parse(valueJSON);
+    setValueString(
+      isTimeseries ? partitionsToText(valueArr, partitionNameArr) : valueArr.join(', '),
+    );
+  }, [valueJSON, partitionNameJSON, isTimeseries]);
```

Comment:
It's possible that the assets being represented by this input all have different partition definitions. In that case the valueJSON would in fact meaningfully change. Maybe we only want to do this useEffect once when we initially get a non empty values array instead? wdyt @hellendag